packages/react-components/src/components/data-providers/tsp-data-provider.ts

```diff
@@ -61,12 +62,32 @@ export class TspDataProvider {
                 rows.push(this.getRowModel(row, chartStart, rowId, entry));
             }
         });
-
+        const annotations: Map<number, TimelineChart.TimeGraphAnnotation[]> = new Map();
+        const annotationsResponse = (await this.client.fetchAnnotations(this.traceUUID,
```

Comment:
I tried it with Matthew's patch https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/173517 for ust annotations in kernel trace and for some reason, the server returned a 500 and i'm not sure where the bug is exactly, but the view returns nothing. You should probably the response status, or at least surrount the annotation block with try-catch to avoid an empty view when annotations return errors

Reply:
Have you figured out where the 500 Internal Server Error comes from? It doesn't look like one of the possible failures in the trace-server.

We plan to address robustness and error handling in a later patch, if this is OK.

Reply:
> Have you figured out where the 500 Internal Server Error comes from? It doesn't look like one of the possible failures in the trace-server.

Not, but I did not investigate!

> We plan to address robustness and error handling in a later patch, if this is OK.

Sure, but you could just put a try catch around those lines to avoid empty views for now. Real error handling can wait.

Reply:
I got the patch to work... after some finagling... lots of finagling... my take:

We need to call this.client.fetchCategories() first. Even if we don't use the result, at least to get the dp into the right state.

Reply:
Also: 
![annotations](https://user-images.githubusercontent.com/419535/102270335-dc592b00-3eeb-11eb-9f66-cf608ad1f91f.png)